File: frontend/src/app/representations/club-representation.ts
import { CategoryRepresentation } from "./category-representation";
import { MemberRepresentation } from "./member-representation";


export interface ClubRepresentation {
  clubId?: number;
  name: string;
  location: string;
  foundedYear: number;
  contactEmail: string;
  contactPhone: string;
  bankAccount: string;
  bankName: string;
  categories: CategoryRepresentation[];      
  members: MemberRepresentation[];          
  files: ClubFile ; 
}



export interface ClubFile {
  id: number;
  licenseUrl: string;
  logoUrl: string;

}

File: backend/src/main/java/com/MaFederation/MaFederation/model/ClubRepresentation.java
package com.MaFederation.MaFederation.model;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder

public class ClubRepresentation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer clubID;

    private String name;
    private String location;
    private Integer foundedYear;

    private String contactEmail;
    private String contactPhone;

    private String bankAccount;
    private String bankName;






}
